File: Barracks.cpp
#include "Barracks.hpp"
#include "units.hpp"

bool
Barracks::create(UnitType u) {
	Unit *unit = NULL;

	for (int i = 0; i < sizeof(units) / sizeof(UnitData); i++)
		if ((units[i].type == u) && (units[i].where == bt_Barracks)) {
			if (this->owner->cost(units[i].gold, units[i].wood))
				unit = units[i].make(this->owner);
			break;
		}

	if (!unit)
		return false;

	unit->place(this->map, this->x, this->y);
	this->owner->addUnit(unit);
	return true;
}

File: ruby/Barracks.rb
# not ruby yet!

require './Building.rb'

class Barracks < Building
public:
	Barracks(Player *owner) : Building("Barracks", owner) {
		@type = bt_Barracks
		@hitpoints = 800
		owner.addScore(80)
	}

	static Building *create(Player *owner) { return new Barracks(owner); }
end

